dashboard/app/models/follower.rb

```diff
@@ -23,6 +23,7 @@ class Follower < ApplicationRecord
   belongs_to :section
   has_one :user, through: :section
   belongs_to :student_user, foreign_key: "student_user_id", class_name: 'User'
+  has_many :code_review_group_members, dependent: :delete_all
```

Comment:
could a user be in multiple groups if they are in multiple sections (for example, took csp then csa in the situation where csp adopts code review)? I don't think we can expect the old groups to be deleted at the end of the year, just expired.

Reply:
A `User` could, but a `Follower` should not be.

Reply:
...right?

Reply:
I updated this. In order to get it to work I had to set a primary key on `code_review_group_members`, which I set to `follower_id`, lmk if that should change.

Reply:
That makes sense to me logically -- is it the association or the dependent destroy or something else that doesn't work if don't include the primary key? I don't know if Rails will do anything weird if we're calling a field a primary key that isn't a primary key from a database perspective

Reply:
it's the `dependent: :delete` that doesn't work (weirdly it does work for has_many and delete_all).

Reply:
Got it -- could you create a follow-up task to migrate to add a primary key on `follower_id`? I'm seeing everywhere else that we explicitly set a primary key attribute there's an actual DB primary key for that field.

Reply:
I attempted to make a follow up to add a primary key, but it seems like the way to do it after the fact is to use the primary_key method. I think I'll leave as-is for now unless you think it'll cause issues.

Reply:
Ok I think I was worried not having a db-level primary key might cause some performance issues, since it looks like primary keys are indexed by default:

https://stackoverflow.com/questions/1071180/is-the-primary-key-automatically-indexed-in-mysql

but since we're already indexing on follower_id I think we should be fine. We don't get the uniqueness constraint at the DB level I guess then.